src/__tests__/UserProfile.test.js

```diff
@@ -10,18 +10,23 @@ import { saveUserInfo } from '../mocks/userSliceMock';
 import UserProfile from '../components/UserProfile';
 import sendAPI from '../utils/sendAPI';
 
-const modalRoot = document.createElement('div');
-modalRoot.setAttribute('id', 'backdrop-root');
-document.body.appendChild(modalRoot);
-
 describe('Profile', () => {
-  beforeEach(async () => {
-    const store = configureStore({
-      reducer: { user: userSliceMock },
-    });
+  const store = configureStore({
+    reducer: { user: userSliceMock },
+  });
 
-    const dispatch = jest.spyOn(store, 'dispatch');
+  beforeAll(() => {
+    const rootMarkup = `
+      <div id="backdrop-root"></div>
+      <div id="overlay-root"></div>
+    `;
+
+    document.body.insertAdjacentHTML('afterbegin', rootMarkup);
+  });
```

Comment:
기존에서 바꾼 이유 여쭤봐도 될까요??
https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML
보긴했습니다! 
그런데 기존코드가 
https://testing-library.com/docs/example-react-modal/
예제를 따라간거라 이유가 궁금합니다

Reply:
포탈 관련해서 인거 같긴한데 이유 알려주세요!

Reply:
DOM을 만드는 과정 코드를 생략하고 바로 넣을 수 있어서 가독성이 좋다고 생각해서 이렇게 바꿨습니다.

innerHTML을 쓸 수도 있는데, 제가 찾아본 결과로는 DOM의 일부만 바꿀때 innerHTML은 HTML 전체를 파싱해서 다른 점을 찾아서 넣어서 해당 위치에 DOM을 끼워넣는 insertAdjacentHTML보다 느린걸로 알고 있습니다.

Reply:
예제의 경우 백드랍이랑 모달을 하나의 div로 묶어서 사용해서 root 엘레멘트가 하나만 있어도 됐지만, 저희 프로젝트의 경우 백드랍을 여러 모달에서 재활용하는지라 백드랍이 모달이랑 분리되어 있어서 HTML 내부에 포탈 도착 위치 역할을 할 div가 두개가 필요합니다.

Reply:
overlay-root div 가 없었던게 어제의 문제였네요
해결방식이 잘못되었던것 같습니다 
슬랙으로 소통한게 좋았던거 같아요

MDN에서 가져왔는데 유빈님 설명이랑 같습니다
To insert the HTML into the document rather than replace the contents of an element, use the method [insertAdjacentHTML()](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML).

하나 배워가네요
좋은 수정 좋은 설명인것 같습니다 ㅎㅎ

Reply:
아 저도 하나 배워갑니다!👍

Reply:
Key에 관련해서는 전에는 UUID를 많이 사용했었는데 

Index Key를 사용하면 안되는 이유 : https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Random Key를 사용하면 안되는 이유(reconciliation) : https://stackoverflow.com/questions/52196127/different-ways-to-add-a-key-to-jsx-element-in-loop-in-react/52341602#52341602
https://stackoverflow.com/questions/39549424/how-to-create-unique-keys-for-react-elements

좋아보여서 공유해드려요